### PR TITLE
3.1.0 API changes page

### PR DESCRIPTION
**Pull request by dstansby**
This will hopefully turn into the final 3.1.0 API changes page.

In addition to merging the API changes I have slightly changed how they are shown. Instead of putting all the text at the bottom of `/api/api_changes.html`, I have created a separate `3.1.0` page with the rest of the version changes, and just included it in the `/api/api_changes.html` page. This means

- No having to manually move these changes in the next version to a new file
- `/api/api_changes_old.html` contains a complete (including latest version) list of changes

### CODE REVIEW

**Comment by tacaswell** [comment]
@dstansby Hope you do not mind I pushed some commits, should have asked first :disappointed:

**Comment by dstansby** [comment]
No problem at all!

**Comment by dstansby** [comment]
Reorganising this into something coherent is done now, if anyone has the energy to proof read it please do... @matplotlib/developers

File: doc/api/prev_api_changes/api_changes_3.1.0.rst [hunk 1]
```diff
@@ -0,0 +1,987 @@
...
+  Future versions will raise an error.
+
+Changes in parameter names
+--------------------------
+
+- The ``arg`` parameter to `matplotlib.use` has been renamed to ``backend``.
+
+This will only affect cases where that parameter has been set
+as a keyword argument. The common usage pattern as a positional argument
+``matplotlib.use('Qt5Agg')`` is not affected.
```
Comment (ImportanceOfBeingErnest) [inline doc/api/prev_api_changes/api_changes_3.1.0.rst]: Does this paragraph need to be indented to match the enumeration pattern?

File: doc/api/prev_api_changes/api_changes_3.1.0.rst [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+
+- ``Spine.is_frame_like``
+
+This has not been used in the codebase since its addition in 2009.
+
+- ``axis3d.Axis.get_tick_positions``
+
+This has never been used internally, there is no equivalent method exists on
+the 2D Axis classes, and despite the similar name, it has a completely
+different behavior from the 2D Axis' `axis.Axis.get_ticks_position` method.
```
Comment (ImportanceOfBeingErnest) [inline doc/api/prev_api_changes/api_changes_3.1.0.rst]: Indentation (s.a.)?

**Comment by tacaswell** [comment]
@dstansby I intend to rebase and force push this branch (to pick up #13908 to check the links are correct).

**Comment by tacaswell** [comment]
If this builds cleanly I am going to merge this, we can continue to work on it during the RC cycle.

Again, thank you @dstansby !

**Pull request merged**